pkg/provision/engine.go

```diff
@@ -151,7 +151,7 @@ func (e *Engine) Run(ctx context.Context) error {
 				triggeredCleanupOnBoot = true
 			}
 
-		case <-cleanUp:
+		case v := <-cleanUp:
```

Comment:
I don't think you need to assign the value, it's an empty struct.
```suggestion
		case  <-cleanUp:
```